tests/test_iam/test_iam.py

```diff
@@ -2879,3 +2880,661 @@ def test_delete_role_with_instance_profiles_present():
+        role_config_query.backends["global"]
+        .roles[role_with_inline_policy["id"]]
+        .to_config_dict()
+    )
+    assert (
+        role_with_inline_policy_config["arn"]
+        == "arn:aws:iam::123456789012:role/role_with_inline_policy"
+    )
+    assert role_with_inline_policy_config["configuration"]["rolePolicyList"] == [
+        {
+            "policyName": "inline_policy",
+            "policyDocument": parse.quote(json.dumps(basic_policy)),
+        }
+    ]
+
+
+@mock_iam
+@mock_config
+def test_role_config_client():
+    from moto.iam.models import ACCOUNT_ID
+    from moto.iam.utils import random_resource_id
+
+    iam_client = boto3.client("iam", region_name="us-west-2")
+    config_client = boto3.client("config", region_name="us-west-2")
+
+    account_aggregation_source = {
+        "AccountIds": [ACCOUNT_ID],
+        "AllAwsRegions": True,
+    }
+
+    config_client.put_configuration_aggregator(
+        ConfigurationAggregatorName="test_aggregator",
+        AccountAggregationSources=[account_aggregation_source],
+    )
+
+    result = config_client.list_discovered_resources(resourceType="AWS::IAM::Role")
+    assert not result["resourceIdentifiers"]
+
+    role_id = iam_client.create_role(
+        Path="/",
+        RoleName="mytestrole",
+        Description="mytestrole",
+        AssumeRolePolicyDocument=json.dumps("{ }"),
+    )["Role"]["RoleId"]
+
+    iam_client.create_role(
+        Path="/",
+        RoleName="mytestrole2",
+        Description="zmytestrole",
+        AssumeRolePolicyDocument=json.dumps("{ }"),
+    )
+
+    # Test non-aggregated query: (everything is getting a random id, so we can't test names by ordering)
+    result = config_client.list_discovered_resources(
+        resourceType="AWS::IAM::Role", limit=1
+    )
+    first_result = result["resourceIdentifiers"][0]["resourceId"]
+    assert result["resourceIdentifiers"][0]["resourceType"] == "AWS::IAM::Role"
+    assert len(first_result) == len(random_resource_id())
+
+    # Test non-aggregated pagination
+    assert (
+        config_client.list_discovered_resources(
+            resourceType="AWS::IAM::Role", limit=1, nextToken=result["nextToken"]
+        )["resourceIdentifiers"][0]["resourceId"]
+    ) != first_result
+
+    # Test aggregated query: (everything is getting a random id, so we can't test names by ordering)
+    agg_result = config_client.list_aggregate_discovered_resources(
+        ResourceType="AWS::IAM::Role",
+        ConfigurationAggregatorName="test_aggregator",
+        Limit=1,
+    )
+    first_agg_result = agg_result["ResourceIdentifiers"][0]["ResourceId"]
+    assert agg_result["ResourceIdentifiers"][0]["ResourceType"] == "AWS::IAM::Role"
+    assert len(first_agg_result) == len(random_resource_id())
+    assert agg_result["ResourceIdentifiers"][0]["SourceAccountId"] == ACCOUNT_ID
+    assert agg_result["ResourceIdentifiers"][0]["SourceRegion"] == "global"
+
+    # Test aggregated pagination
```

Comment:
Try adding in a test with filtering and pagination. This should result in errors without the filtering logic implemented.